irohad/ametsuchi/impl/flat_file/flat_file.cpp

```diff
@@ -16,132 +16,84 @@
-  std::sprintf(&new_id[0], "%016u", id);
-  return new_id;
-}
-
-/**
- * Convert string to identifier
- * @param name - string for conversion
- * @return numeric identifier
- */
-Identifier name_to_id(const std::string &name) {
-  std::string::size_type sz;
-  return static_cast<Identifier>(std::stoul(name, &sz));
-}
-
-/**
- * Check if file exists
- * @param name - full path to file
- * @return true, if exists
- */
-bool file_exist(const std::string &name) {
-  struct stat buffer {};
-  return stat(name.c_str(), &buffer) == 0;
-}
-
-/**
- * Remove file from folder
- * @param dump_dir - target dir
- * @param id - identifier of file
- */
-void remove(const std::string &dump_dir, std::string filename) {
-  auto f_name = dump_dir + SEPARATOR + filename;
-
-  if (std::remove(f_name.c_str()) != 0) {
-    logger::log("FLAT_FILE")
-        ->error("remove({}, {}): error on deleting file", dump_dir, filename);
+namespace {
+  const uint32_t DIGIT_CAPACITY = 16;
+
+  /**
+   * Convert id to a string representation. The string representation is always
+   * DIGIT_CAPACITY-character width regardless of the value of `id`.
+   * If the length of the string representation of `id` is less than
+   * DIGIT_CAPACITY, then the returned value is filled with leading zeros.
+   *
+   * For example, if str_rep(`id`) is "123", then the returned value is
+   * "0000000000000123".
+   *
+   * @param id - for conversion
+   * @return string repr of identifier
+   */
+  std::string id_to_name(Identifier id) {
+    std::ostringstream os;
+    os << std::setw(DIGIT_CAPACITY) << std::setfill('0') << id;
+    return os.str();
   }
-}
 
-/**
- * Checking consistency of storage for provided folder
- * If some block in the middle is missing all blocks following it are deleted
- * @param dump_dir - folder of storage
- * @return - last available identifier
- */
-nonstd::optional<Identifier> check_consistency(const std::string &dump_dir) {
-  auto log = logger::log("FLAT_FILE");
-
-  Identifier tmp_id = 0u;
-  if (dump_dir.empty()) {
-    log->error("check_consistency({}), not directory", dump_dir);
-    return nonstd::nullopt;
-  }
-  // Directory iterator:
-  struct dirent **namelist;
-  auto status = scandir(dump_dir.c_str(), &namelist, nullptr, alphasort);
-  if (status < 0) {
-    log->error("check_consistency({}), scandir error: {}", dump_dir, status);
-    return nonstd::nullopt;
-  }
-  if (status < 3) {
-    log->info("check_consistency({}), directory is empty", dump_dir);
-    auto n = static_cast<uint32_t>(status);
-    for (auto j = 0u; j < n; ++j) {
-      free(namelist[j]);
+  /**
+   * Checking consistency of storage for provided folder
+   * If some block in the middle is missing all blocks following it are deleted
+   * @param dump_dir - folder of storage
+   * @return - last available identifier
+   */
+  nonstd::optional<Identifier> check_consistency(const std::string &dump_dir) {
+    auto log = logger::log("FLAT_FILE");
+
+    if (dump_dir.empty()) {
```

Comment:
We should use [canonical](http://www.boost.org/doc/libs/1_65_0/libs/filesystem/doc/reference.html#canonical) paths everywhere instead of `std::string`. But this is out of scope of this PR.

Reply:
The validation part still seems like a valid point, even it the argument is still `std::string`.

Reply:
Whether or not the validation is redundant does not depends on how the `check_consistency` is called, but the contract of `check_consistency` determines it.

For example, on one hand, if the precondition of `check_consistency` is `boost::filesystem::is_directory(dump_dir)`, then the check is redundant.

On the other hand, if the contract of `check_consistency` promises to return the `nullopt` when `!boost::filesystem::is_directory(dump_dir)`, then we need the check.

I didn't want to change the semantics of the function in this PR, so I left the check unchanged.